pipert2/core/base/routine.py

```diff
@@ -104,23 +104,49 @@ def _extended_run(self) -> None:
         """
         raise NotImplementedError
 
-    def setup(self) -> None:
+    def setup(self) -> None:  # TODO - Can be auto created with annotation
         """An initial setup before running
+        The user supposed to implement this method
 
         """
 
         pass
 
-    def cleanup(self) -> None:
+    def cleanup(self) -> None:  # TODO - Can be auto created with annotation
+        """The final method that ends the routine execution
+        The user supposed to implement this method
+
+        """
+
+        pass
+
+    def _base_setup(self) -> None:
+        """An initial setup before running
+
+        """
+
+        # Currently we don't have any additional setup
```

Comment:
I think we should remove this function for now and if we're going to need it we will add it like base_cleanup. (I smell little YAGNI)